src/main/java/org/opensearch/security/dlic/rest/api/AccountApiAction.java

```diff
@@ -183,47 +237,106 @@ protected void handleGet(RestChannel channel, RestRequest request, Client client
     @Override
     protected void handlePut(RestChannel channel, final RestRequest request, final Client client, final JsonNode content) throws IOException {
         final User user = threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
-        final String username = user.getName();
         final SecurityDynamicConfiguration<?> internalUser = load(CType.INTERNALUSERS, false);
 
-        if (!internalUser.exists(username)) {
-            notFound(channel, "Could not find user.");
+        if (user == null){
```

Comment:
This two are different changes. Both the checks should be present. 

This is a calling user - `if (user == null)`
This is to check if the calling user is present in the index or not